l1-contracts/contracts/bridge/asset-router/L1AssetRouter.sol

```diff
@@ -0,0 +1,449 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+// solhint-disable reason-string, gas-custom-errors
+
+// import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable-v4/access/Ownable2StepUpgradeable.sol";
+// import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable-v4/security/PausableUpgradeable.sol";
+
+import {IERC20} from "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+
+import {IL1AssetRouter} from "./IL1AssetRouter.sol";
+import {IAssetRouterBase} from "./IAssetRouterBase.sol";
+import {AssetRouterBase} from "./AssetRouterBase.sol";
+
+import {IL2Bridge} from "../interfaces/IL2Bridge.sol";
+// import {IL2BridgeLegacy} from "./interfaces/IL2BridgeLegacy.sol";
+import {IL1AssetHandler} from "../interfaces/IL1AssetHandler.sol";
+import {IAssetHandler} from "../interfaces/IAssetHandler.sol";
+import {IL1Nullifier} from "../interfaces/IL1Nullifier.sol";
+// import {IL1NativeTokenVault} from "../ntv/IL1NativeTokenVault.sol";
+import {INativeTokenVault} from "../ntv/INativeTokenVault.sol";
+// import {IL1SharedBridgeLegacy} from "./interfaces/IL1SharedBridgeLegacy.sol";
+import {IL2SharedBridgeLegacyFunctions} from "../interfaces/IL2SharedBridgeLegacyFunctions.sol";
+
+import {ReentrancyGuard} from "../../common/ReentrancyGuard.sol";
+import {DataEncoding} from "../../common/libraries/DataEncoding.sol";
+import {AddressAliasHelper} from "../../vendor/AddressAliasHelper.sol";
+import {TWO_BRIDGES_MAGIC_VALUE, ETH_TOKEN_ADDRESS} from "../../common/Config.sol";
+// import {L2_NATIVE_TOKEN_VAULT_ADDRESS} from "../../common/L2ContractAddresses.sol";
+import {Unauthorized, ZeroAddress, TokenNotSupported, AddressAlreadyUsed} from "../../common/L1ContractErrors.sol";
+import {L2_ASSET_ROUTER_ADDR} from "../../common/L2ContractAddresses.sol";
+
+import {IBridgehub, L2TransactionRequestTwoBridgesInner, L2TransactionRequestDirect} from "../../bridgehub/IBridgehub.sol";
+
+// import {BridgeHelper} from "./BridgeHelper.sol";
+
+import {IL1AssetDeploymentTracker} from "../interfaces/IL1AssetDeploymentTracker.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @dev Bridges assets between L1 and ZK chain, supporting both ETH and ERC20 tokens.
+/// @dev Designed for use with a proxy for upgradability.
+contract L1AssetRouter is
```

Comment:
if this contract is supposed to be the upgraded version of the l1sharedbridge, then it is not compatible with the storage layout

Reply:
L1SharedBridge will be upgraded to L1Nullifier, and L1AssetRouter will be a new contract.